nebullvm/inference_learners/onnx_mlir.py

```diff
@@ -0,0 +1,211 @@
+import copy
+import os
+import shutil
+import sys
+import warnings
+from abc import ABC
+from pathlib import Path
+from typing import Dict, Generator, List, Tuple, Type, Union
+
+import cpuinfo
+import numpy as np
+import tensorflow as tf
+import torch
+from nebullvm.base import DeepLearningFramework, ModelParams
+from nebullvm.config import ONNX_MLIR_FILENAMES
+from nebullvm.inference_learners.base import (BaseInferenceLearner,
+                                              LearnerMetadata,
+                                              PytorchBaseInferenceLearner,
+                                              TensorflowBaseInferenceLearner)
+
+try:
+    # Set the ONNX_MLIR_HOME as the environment variable and append in the path,
+    # directory path where the MLIR is built
+    MLIR_INSTALLATION_ROOT = Path.home()
+
+    os.environ['ONNX_MLIR_HOME'] = os.path.join(
+        MLIR_INSTALLATION_ROOT,
+        'onnx-mlir',
+        'build',
+        'Debug',
+    )
+
+    sys.path.append(
+        os.path.join(
+            os.environ.get('ONNX_MLIR_HOME', ''),
+            'lib',
+        )
+    )
+    import PyRuntime
+except ImportError:
+    warnings.warn(
+        "No valid onnxruntime installation found. Trying to install it..."
+    )
+    from nebullvm.installers.installers import install_onnx_mlir
+
+    install_onnx_mlir(
```

Comment:
I think we can disable the ONNX-MLIR auto-installer for the time being. Since it's a rather long install, I would suggest treating it similarly to TVM, i.e. allowing the user to install the compiler only by explicitly running the `install_onnx_mlir` function